lmfit 5.1 (new formula)

T045T opened the pull request:
Added a formula for lmfit ( http://apps.jcns.fz-juelich.de/doku/sc/lmfit ).

No test block because I wasn't sure about compiling a test program in a test - both how to best do it and whether it should be done at all.

tdsmith commented on lmfit.rb:
```diff
@@ -0,0 +1,103 @@
+  test do
+    (testpath/"test.c").write <<-EOS
+    #include "lmcurve.h"
+    #include "lmmin.h"
```
Can you indent the test code another 2 spaces?

tdsmith commented:
The test looks good! If you want you can install the demo from the tarball with e.g. `(share/"lmfit").install Dir["demo/*"]` or whatever and then build it from there in the test block.

dpo commented:
+1 for the demo.

T045T commented:
I changed the formula to install the demo sources - since they're supposed to demonstrate how the library works, I thought this made more sense than copying the binaries.

I also added a caveat to inform the user about those demo files. Is that okay?

dpo commented:
Great. Thank you for your contribution!
